src/main/java/uk/gov/pay/api/filter/LoggingMDCRequestFilter.java

```diff
@@ -11,18 +11,26 @@
 import java.security.Principal;
 import java.util.Optional;
 
+import static org.apache.commons.lang3.StringUtils.EMPTY;
 import static uk.gov.pay.logging.LoggingKeys.GATEWAY_ACCOUNT_ID;
-import static uk.gov.pay.logging.LoggingKeys.PAYMENT_EXTERNAL_ID;
 import static uk.gov.pay.logging.LoggingKeys.MANDATE_EXTERNAL_ID;
+import static uk.gov.pay.logging.LoggingKeys.PAYMENT_EXTERNAL_ID;
 import static uk.gov.pay.logging.LoggingKeys.REFUND_EXTERNAL_ID;
+import static uk.gov.pay.logging.LoggingKeys.REMOTE_ADDRESS;
 
 @Provider
 @Priority(Priorities.USER)
 public class LoggingMDCRequestFilter implements ContainerRequestFilter {
 
     @Override
     public void filter(ContainerRequestContext requestContext) {
-        MDC.put(GATEWAY_ACCOUNT_ID, getAccountId(requestContext));
+        Optional<Account> mayBeAccount = getAccount(requestContext);
+        MDC.put(GATEWAY_ACCOUNT_ID, mayBeAccount.map(Account::getName).orElse(EMPTY));
+        MDC.put("token_link", mayBeAccount.map(Account::getTokenLink).orElse(EMPTY));
```

Comment:
as this is logged across both public api and public auth (and could be potentially useful over in self service logs), it's potentially a good candidate for a shared key?